### DISCUSSION TITLE
change language

### DISCUSSION THREAD

**Discussion by zarevskaya** [mediacms-io/mediacms#932] [General]
**Describe the issue**
I want to set the website in french
mediacms last version, no issue.

Hello, how is possible to translate mediacms in french?

I had added fr like this (it's the good way?):

![Screenshot_20231202_211740](https://github.com/mediacms-io/mediacms/assets/60230221/a178ffbd-bc4a-45f5-bc01-38a9c81740af)

But there is no change. I have doubts about what I did...
Any hint?

Thank you :)
Oh, amazing app! I love it!

**Comment by mgogoulos** [comment]
Translation is not yet ready, it's almost ready on a branch but not fully ready

**Reply by zarevskaya** [reply]
Thanks for the answer.

**Reply by arman-media** [reply]
> La traduction n'est pas encore prête, elle est presque prête sur une branche mais pas entièrement prête

Bonjour, est-ce qu'aujourd'hui cela fonctionne ?

**Comment by partonen** [comment]
Hello, please tell me when the translation functionality will be completed? Or is it possible to launch a project from a branch and it will be operational?

**Reply by dnnrwttr** [reply]
Yes, need it urgently too :)

**Comment by mgogoulos** [comment]
They are now included in the latest release! Check it out on https://demo.mediacms.io/ and update!!!

Also if you want to help improving them, checkout https://github.com/mediacms-io/mediacms/blob/main/docs/admins_docs.md#20-translations

**Comment by zarevskaya** [comment]
Hello, I don't understand what code I need to use for french? There is a list anywhere?
![Screenshot_20241004_132742](https://github.com/user-attachments/assets/4b996d30-0dcc-432c-960a-c9a8b8add813)

I added this line but it's always in English.

I can't select french like the MediaCMS demo :

![Screenshot_20241004_133304](https://github.com/user-attachments/assets/c0f8b11b-1c9d-4619-8b86-01726a98b3a0)


MediaCMS is updated. New installation.

**Reply by partonen** [reply]
Hello, I have a similar problem.

**Comment by mgogoulos** [comment]
@partonen @zarevskaya french is already an included language, you just have to checkout the latest version. Did you upgrade? If so, check settings.py, it already contains french

**Comment by zarevskaya** [comment]
It's a fresh install. I'm starting the installation again, it's quick. I can see the frontend_translations folder.

**Comment by zarevskaya** [comment]
Ok... Was my cache... Sorry! Amazing update! Thanks!

**Reply by partonen** [reply]
This is not a cache, try opening the site in incognito mode and the language will always be English.

**Reply by zarevskaya** [reply]
For me it was the cache. In incognito mode that work fine. 

Do you have frontend_translations in /files?

**Reply by partonen** [reply]
Yes, there are frontend_translations files.

**Reply by mgogoulos** [reply]
I also got to see the cached version initially, i'll open a ticket to address the caching problems on css/js files. 

As far as the default language is related, it's Django's behavior how it is handled - https://docs.djangoproject.com/en/5.1/topics/i18n/translation/#how-django-discovers-language-preference


You can change the language by selecting the 'Language' on the left sidebar, and see the available options. This is a v1 feature, the languages selection would probably have to be moved in a more visible place etc

**Comment by partonen** [comment]
Also in the settings.py file the variable LANGUAGE_CODE is repeated twice at the beginning and at the end, which is also a little confusing.

**Reply by mgogoulos** [reply]
nice catch, I fixed this now
